### PR TITLE
PSP-9341 : Sub-file interest type

### PR DESCRIPTION
**Pull request by eddherrera**
image

### CODE REVIEW

File: source/frontend/src/constants/API.ts [hunk 1]
```diff
@@ -134,6 +134,7 @@ export const HISTORICAL_NUMBER_TYPES = 'PimsHistoricalFileNumberType';
 export const LEASE_CHECKLIST_SECTION_TYPES = 'PimsLeaseChklstSectionType';
 export const LEASE_CHECKLIST_ITEM_STATUS_TYPES = 'PimsChklstItemStatusType';
 export const LEASE_PAYMENT_CATEGORY_TYPES = 'PimsLeasePaymentCategoryType';
+export const SUBFILE_INTERETST_TYPES = 'PimsSubfileInterestType';
```
Comment (devinleighsmith) [inline source/frontend/src/constants/API.ts:137]: NIT: SUBFILE_INTERETST_TYPES -> SUBFILE_INTEREST_TYPES

File: source/frontend/src/features/mapSideBar/acquisition/add/AddAcquisitionForm.tsx [hunk 2]
```diff
@@ -133,9 +136,23 @@ const AddAcquisitionDetailSubForm: React.FC<{
   const acquisitionTypes = getOptionsByType(API.ACQUISITION_TYPES);
   const acquisitionPhysFileTypes = getOptionsByType(API.ACQUISITION_PHYSICAL_FILE_STATUS_TYPES);
   const acquisitionFundingTypes = getOptionsByType(API.ACQUISITION_FUNDING_TYPES);
+  const subfileInterestTypes = getOptionsByType(API.SUBFILE_INTERETST_TYPES);
 
   const isSubFile = exists(parentId) && isValidId(parentId);
 
+  const {
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/acquisition/add/AddAcquisitionForm.tsx:143]: Hmm, This is just a re-order of the existing code right?

File: source/frontend/src/features/mapSideBar/acquisition/tabs/fileDetails/detail/models.ts [hunk 3]
```diff
@@ -1,26 +1,38 @@
...
 
 export class DetailAcquisitionFile {
+  isSubFile: boolean;
   fileName?: string;
   legacyFileNumber?: string;
   assignedDate?: string;
   deliveryDate?: string;
   acquisitionPhysFileStatusTypeDescription?: string;
   acquisitionTypeDescription?: string;
+  subfileInteresttypeDescription: string | null = null;
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/acquisition/tabs/fileDetails/detail/models.ts:14]: nit: consistency of capitalization between this line and the next.

File: source/frontend/src/features/mapSideBar/acquisition/tabs/fileDetails/detail/models.ts [hunk 4]
```diff
@@ -1,26 +1,38 @@
...
     detail.fileName = model?.fileName ?? undefined;
     detail.legacyFileNumber = model?.legacyFileNumber ?? undefined;
     detail.assignedDate = model?.assignedDate ?? undefined;
     detail.deliveryDate = model?.deliveryDate ?? undefined;
     detail.acquisitionPhysFileStatusTypeDescription =
       model?.acquisitionPhysFileStatusTypeCode?.description ?? undefined;
     detail.acquisitionTypeDescription = model?.acquisitionTypeCode?.description ?? undefined;
+
+    if (detail.isSubFile) {
+      detail.subfileInteresttypeDescription = model.subfileInterestTypeCode.description;
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/acquisition/tabs/fileDetails/detail/models.ts:33]: possible NPE here? even if the subfileInterestTypeCode shouldn't be null, I'd still guard against null here.

Comment (devinleighsmith) [reply source/frontend/src/features/mapSideBar/acquisition/tabs/fileDetails/detail/models.ts:33]: (for etl'd data most likely).

File: source/frontend/src/features/mapSideBar/acquisition/add/AddAcquisitionForm.tsx [hunk 5]
```diff
@@ -82,6 +83,8 @@ export const AddAcquisitionForm = React.forwardRef<
       innerRef={ref}
       initialValues={initialValues}
       validationSchema={validationSchema}
+      validateOnChange={false}
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/acquisition/add/AddAcquisitionForm.tsx:86]: What was the reason for this change?

File: source/frontend/src/features/mapSideBar/acquisition/tabs/fileDetails/update/UpdateAcquisitionForm.test.tsx [hunk 6]
```diff
@@ -235,4 +241,68 @@ describe('UpdateAcquisitionForm component', () => {
     expect(validationSchema).toBeCalled();
     expect(onSubmit).toHaveBeenLastCalledWith(initialValues, expect.anything());
   });
+
+  describe('Sub-interest files', () => {
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/acquisition/tabs/fileDetails/update/UpdateAcquisitionForm.test.tsx:245]: Thanks for adding these tests, looks good.

**Review by devinleighsmith — CHANGES_REQUESTED** [review]
I'd like to see the possible NPE be corrected, otherwise looks good.

File: source/frontend/src/features/mapSideBar/acquisition/add/AddAcquisitionForm.tsx (hunk 2 above)
Comment (eddherrera) [reply source/frontend/src/features/mapSideBar/acquisition/add/AddAcquisitionForm.tsx:143]: Yes

File: source/frontend/src/features/mapSideBar/acquisition/tabs/fileDetails/detail/models.ts (hunk 3 above)
Comment (eddherrera) [reply source/frontend/src/features/mapSideBar/acquisition/tabs/fileDetails/detail/models.ts:14]: Updated

File: source/frontend/src/features/mapSideBar/acquisition/add/AddAcquisitionForm.tsx (hunk 5 above)
Comment (eddherrera) [reply source/frontend/src/features/mapSideBar/acquisition/add/AddAcquisitionForm.tsx:86]: When changing the Type to 'Other' it's triggering the OtherDescription validation before even entering the value and/or submitting.

File: source/frontend/src/features/mapSideBar/acquisition/tabs/fileDetails/detail/models.ts (hunk 4 above)
Comment (eddherrera) [reply source/frontend/src/features/mapSideBar/acquisition/tabs/fileDetails/detail/models.ts]: updated

File: source/frontend/src/constants/API.ts (hunk 1 above)
Comment (eddherrera) [reply source/frontend/src/constants/API.ts:137]: Updated

**Review by devinleighsmith — APPROVED** [review]

**Pull request merged**